### PR TITLE
Improve assets order

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-sdk/src/core/observers/BalanceObserver.ts [hunk 1]
```diff
@@ -44,21 +44,25 @@ export class BalanceObserver {
...
+    }
+    return tokenBalances;
   }
 
   private isInitialCall() {
     return this.lastTokenBalances.length === 0;
   }
 
-  private async getTokensWithUpdatedBalances(addresses: string[]) {
+  private async getTokenBalances() {
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/core/observers/BalanceObserver.ts:63]: rename to `getInitialTokenBalances`

Comment (Sajma27) [reply universal-login-sdk/src/core/observers/BalanceObserver.ts]: Done

File: universal-login-sdk/src/core/observers/BalanceObserver.ts [hunk 2]
```diff
@@ -44,21 +44,25 @@ export class BalanceObserver {
...
     }
     const changedErc20contracts = await this.getErc20ContractsWithChangedBalances();
-    const addressesToUpdate = [...changedErc20contracts, ETHER_NATIVE_TOKEN.address];
-    const tokensWithoutChanges = this.lastTokenBalances.filter(token => !addressesToUpdate.includes(token.address));
-    const tokensWithUpdatedBalances = await this.getTokensWithUpdatedBalances(addressesToUpdate);
-    return [...tokensWithoutChanges, ...tokensWithUpdatedBalances];
+    const addressesToUpdate = [ETHER_NATIVE_TOKEN.address, ...changedErc20contracts];
+    const tokenBalances: TokenDetailsWithBalance[] = [];
+    for (const token of this.lastTokenBalances) {
+      const balance = addressesToUpdate.includes(token.address) ? await this.balanceChecker.getBalance(this.walletAddress, token.address) : token.balance;
```
Comment (xitronix) [inline universal-login-sdk/src/core/observers/BalanceObserver.ts:53]: extract `lazyGetBalance`

File: universal-login-sdk/src/core/observers/BalanceObserver.ts [hunk 3]
```diff
@@ -44,21 +44,29 @@ export class BalanceObserver {
...
+    const addressesToUpdate = [ETHER_NATIVE_TOKEN.address, ...changedErc20contracts];
+    return this.lazyGetBalance(addressesToUpdate);
   }
 
   private isInitialCall() {
     return this.lastTokenBalances.length === 0;
   }
 
-  private async getTokensWithUpdatedBalances(addresses: string[]) {
+  private async lazyGetBalance(addressesToUpdate: string[]) {
```
Comment (xitronix) [inline universal-login-sdk/src/core/observers/BalanceObserver.ts:58]: Rename to lazyGetBalances

Comment (Sajma27) [reply universal-login-sdk/src/core/observers/BalanceObserver.ts:58]: Done

**Pull request merged**